src/dataflow/src/source/s3.rs

```diff
@@ -731,7 +805,17 @@ impl SourceReader for S3SourceReader {
                     self.id,
                     e
                 );
-                Err(e)
+                match e {
```

Comment:
No retrying happens if this code path is reached. There is just this in the logs:

```
2021-06-09T08:58:15.937766Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-1: Error obtaining body: Error obtaining chunk: error reading a body from connection: end of file before message length reached (9 retries remaining)    
2021-06-09T08:58:16.839016Z  WARN dataflow::source::s3: when reading source 's3-u1/28' (u1/28): Error obtaining chunk: error reading a body from connection: end of file before message length reached
```

and nothing else. So it has "9 retries remaining" but it does not retry any futher. There is no error in the source.

Reply:
Turns out that this retries, but every 30 seconds or so:

```
2021-06-17T08:45:48.687405Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (9 retries remaining)    
2021-06-17T08:46:18.536024Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (9 retries remaining)    
2021-06-17T08:46:39.576908Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (8 retries remaining)    
2021-06-17T08:47:00.614088Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (7 retries remaining)    
2021-06-17T08:47:21.656958Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (6 retries remaining)    
2021-06-17T08:47:42.691541Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (5 retries remaining)    
2021-06-17T08:48:03.731299Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (4 retries remaining)    
2021-06-17T08:48:24.768568Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (3 retries remaining)    
2021-06-17T08:48:45.803819Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (2 retries remaining)    
2021-06-17T08:49:06.840151Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-0: Error during dispatch: connection closed before message completed (1 retries remaining)    
```

So less frequently then e.g. reading from the S3 bucket itself:

```
2021-06-17T10:11:01.732431Z  WARN dataflow::source::s3: Failed to download object: testdrive-test-0/s3/a (attempt 0)    
2021-06-17T10:11:01.875932Z  WARN dataflow::source::s3: Failed to download object: testdrive-test-0/s3/a (attempt 1)    
2021-06-17T10:11:02.144487Z  WARN dataflow::source::s3: Failed to download object: testdrive-test-0/s3/a (attempt 2)    
```